testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+import os
+import shakedown
+import shutil
+import time
+
+import sdk_cmd
+import sdk_hosts
+import sdk_marathon
+import sdk_security
+
+
+log = logging.getLogger(__name__)
+
+KERBEROS_APP_ID = "kdc"
+KERBEROS_IMAGE_NAME = "mesosphere/kdc"
+KERBEROS_KEYTAB_FILE_NAME = "keytab"
+BASE64_ENCODED_KEYTAB_FILE_NAME = "{keytab_name}.base64"
+DCOS_BASE64_PREFIX = "__dcos_base64__"
+LINUX_USER = "core"
+KERBEROS_CONF = "krb5.conf"
+REALM = "LOCAL"
+
+# Note: Some of the helper functions in this module are wrapped in basic retry logic to provide some
+# resiliency towards possible intermittent network failures.
+
+
+def _launch_marathon_app(app_definition):
+    """
+    Launches the marathon app given a marathon app definition.
+    :param app_definition (dict): The app definition to launch the app from.
+    Raises an exception on failure.
+    """
+    log.info("Launching KDC marathon app")
+    rc, msg = sdk_marathon.install_app(app_definition)
+    if not rc:
+        raise RuntimeError("Can't install KDC marathon app: {err}".format(err=msg))
+
+
+@retry(stop_max_attempt_number=3, wait_fixed=2000)
+def _check_kdc_marathon_task_is_running() -> bool:
+    """
+    :return (bool): Indicator of whether the KDC marathon app is running or not. Raises exception if app is not detected
+    after the specified wait & retry period.
+    """
+    log.info("Waiting for app to be running...")
+    shakedown.wait_for_task("marathon", KERBEROS_APP_ID)
+
+
+def _get_task_id() -> str:
+    """
+    :return (str): The ID of the task running the KDC app.
+    """
+    log.info("Getting task id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Task id is {}".format(task["id"]))
+                return task["id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+def _get_framework_id() -> str:
+    """
+    :return (str): The ID of the framework running the KDC app.
+    """
+    log.info("Getting framework id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Framework id is {}".format(task["framework_id"]))
+                return task["framework_id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_id() -> str:
+    """
+    :return (str): The ID of the node running the KDC app.
+    """
+    log.info("Getting host id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
```

Comment:
Same here (see comment w.r.t framework id and task id).

Reply:
Done.